src/util/helpers.py

```diff
@@ -339,3 +339,25 @@ def _remove_export_lines(cr, model, field=None):
     ]
     if to_rem:
         cr.execute("DELETE FROM ir_exports_line WHERE id IN %s", [tuple(to_rem)])
+
+
+def _remove_import_mappings(cr, model, field=None):
+    q = """
+    SELECT id,
+           res_model,
+           STRING_TO_ARRAY(field_name, '/')
+      FROM base_import_mapping
+    """
+    if field:
+        q = cr.mogrify(q + " WHERE field_name ~ %s ", [r"\y{}\y".format(field)]).decode()
+    cr.execute(q)
+    to_rem = [
+        mapping_id
+        for mapping_id, mapping_model, mapping_path in cr.fetchall()
+        if any(
+            x.field_model == model and (field is None or x.field_name == field)
+            for x in resolve_model_fields_path(cr, mapping_model, mapping_path)
+        )
+    ]
+    if to_rem:
+        cr.execute("DELETE FROM base_import_mapping WHERE id IN %s", [tuple(to_rem)])
```

Comment:
This is done in a different new private function, it could be refactored later. I was referring to the extra lines in the main processing logic in `field.py`